src/vim9execute.c

```diff
@@ -4201,6 +4201,41 @@ exec_instructions(ectx_T *ectx)
 		}
 		break;
 
+	    case ISN_BITSHIFT:
+		{
+		    typval_T	*tv1 = STACK_TV_BOT(-2);
+		    typval_T	*tv2 = STACK_TV_BOT(-1);
+		    exprtype_T	exprtype = iptr->isn_arg.op.op_type;
+
+		    if (tv1->vval.v_number < 0 || tv2->vval.v_number < 0)
+		    {
+			SOURCING_LNUM = iptr->isn_lnum;
+			emsg(_(e_bitshift_ops_must_be_postive));
+			goto on_error;
+		    }
+		    if (tv2->vval.v_number > MAX_LSHIFT_BITS)
+		    {
+			SOURCING_LNUM = iptr->isn_lnum;
+			emsg(_(e_bitshift_value_too_large));
+			goto on_error;
+		    }
+
+		    if (exprtype == EXPR_LSHIFT)
+			tv1->vval.v_number =
+				(uvarnumber_T)tv1->vval.v_number << tv2->vval.v_number;
+		    else
+			tv1->vval.v_number =
+				(uvarnumber_T)tv1->vval.v_number >> tv2->vval.v_number;
+
+		    // clear the topmost sign bit
+		    tv1->vval.v_number &=
```

Comment:
I think this to be quite weird, I'd let the user deal with negative numbers rather than making the integers N-1bit wide.

Reply:
This is based on the following comment from Bram:

  Since bitwise operators should not care about a sign bit, I would very
  much prefer ">>" to just make the topmost bit zero.  No need for ">>>"
  then.